spec/unleash/feature_toggle_spec.rb

```diff
@@ -332,7 +332,7 @@
     it 'get_variant_with_matching_override should for user_id:61' do
       # NOTE: Use send method, as we are testing a private method
       context = Unleash::Context.new(user_id: 61)
-      expect(feature_toggle.send(:variant_from_override_match, context)).to have_attributes(
+      expect(feature_toggle.send(:variant_from_override_match, context, feature_toggle.variant_definitions)).to have_attributes(
```

Comment:
Note: line above is perfect. This comment is regarding this file, and not the change above.

We should have at least one extra test in `FeatureToggle` that tests variants within the strategy. Probably within it's own `describe` section....

Also because this is the core of the change.

I know some tests are provided in the client_specification, but I would prefer that we keep some healthy test coverage within the SDK. Specially for the core of the logic.

maybe something like:
```ruby
  describe 'FeatureToggle with variants in the strategy' do
    let(:feature_toggle) do
      Unleash::FeatureToggle.new(
        "name" => "Test.variantsWithinAStrategy",
        "description" => nil,
        "enabled" => true,
        "strategies" => [
          {
            "name" => "flexibleRollout",
            "parameters" => {
              "rollout" => "100",
              "stickiness" => "default",
              "groupId" => "a"
            },
            "variants" => [
              {
                "name" => "VariantAWithinStrategy",
                "weight" => 1,
                "payload" => {
                  "type" => "string",
                  "value" => "variantValueA"
                }
              },
              {
                "name" => "VariantBWithinStrategy",
                "weight" => 1,
                "payload" => {
                  "type" => "string",
                  "value" => "variantValueB"
                }
              }
            ],
            "constraints" => [
              {
                "contextName" => "environment",
                "operator" => "IN",
                "values" => [
                  "dev"
                ]
              }
            ]
          }
        ],
        "variants" => [
          {
            "name" => "variant1",
            "weight" => 50,
            "stickiness" => "default",
            "payload" => {
              "type" => "string",
              "value" => "val1"
            },
            "overrides" => [{
              "contextName" => "userId",
              "values" => ["132", "61"]
            }]
          },
          {
            "name" => "variant2",
            "weight" => 50,
            "stickiness" => "default",
            "payload" => {
              "type" => "string",
              "value" => "val2"
            }
          }
        ],
        "createdAt" => "2019-01-24T10:41:45.236Z"
      )
    end

    it 'should return variantXYZ for user_id:ABC' do
      # ... 
    end
```

Reply:
fixed